tests/unit/notes-dao.test.js

```diff
@@ -0,0 +1,125 @@
+const appRoot = require('app-root-path');
+const chai = require('chai');
+const chaiExclude = require('chai-exclude');
+const config = require('config');
+const fs = require('fs');
+const _ = require('lodash');
+const sinon = require('sinon');
+
+const testData = require('./test-data');
+
+const fsOps = appRoot.require('utils/fs-operations');
+
+chai.use(chaiExclude);
+const { assert } = chai;
+
+sinon.replace(fsOps, 'validateDBPath', () => null);
+sinon.replace(config, 'get', property => testData.mockConfig[property]);
+const notesDAO = appRoot.require('api/v1/db/json/notes-dao');
```

Comment:
This `require` has to come after `fsOps.validateDBPath()` and `config.get()` are replaced so the notesDAO doesn't try to validate the fake database path. Since the logic for validating the database is going to be moved to app.js though, this shouldn't be required. I'll change this.